docs/specification.md

```diff
@@ -311,89 +432,153 @@ available state entry is `D`:
-from the gittuf's guarantees, all commits SHOULD be signed by their authors.
-
-### Making changes available to other users -- `git push`
-
-The RSL is updated when users are ready to push changes for some ref to a remote
-version of the repository. There are some modifications to this workflow from
-what is described in the RSL academic paper. First, the remote's RSL is fetched
-and its entries are evaluated against the current state of the target ref. If
-changes were made to the ref remotely, they need to be incorporated and the
-local changes must be reapplied. Further, any updates to the policy namespace
-must also be applied locally. Once this process is complete (it may take
-multiple passes if the target ref receives a lot of activity on the remote),
-gittuf creates a provisional entry in the local RSL.
-
-This entry is provisional because before the remote can be updated with the new
-status of the target ref and the RSL, gittuf executes the
-[verification workflow](#verifying-changes-made) with the provisional entry.
-This means that prior to changes being pushed, the verification workflow ensures
-the commits and the entry are all valid as per the latest available policy on
-the remote.
-
-If verification passes, the target ref and the RSL entry are pushed to the
-remote, _after_ checking that more entries have not been created on the remote
-when verification was in progress locally. If this is the case, the provisional
-entry is deleted, upstream changes are fetched, and the entire process is
-repeated.
-
-## Recovery Workflows
-
-### Recovering from accidental changes and pushes
-
-There are several scenarios here. If a user makes changes locally and tries to
-push them to the blessed copy, it should be quite easy to detect and reject the
-changes. A pre-receive hook on the server can be employed to ensure the client
-is also sending valid metadata for the set of changes. If not, the operation
-must be terminated.
-
-In situations where server-side hooks cannot be used (or trusted), maintainers
-of the repositories can correct the record for the affected refs and sign new
-RSL entries indicating the correct locations. Clients that employ gittuf are
-always secure as they will reject changes that fail validation.
-
-TODO: evaluate if consecutive state verification fails on clients behind the
-times. Should recovery rewrite non-valid RSL entries? Defeats the purpose?
-
-### Recovering from a developer compromise
-
-If a developer's keys are compromised and used to make changes to the
-repository, maintainers must immediately sign updated policies revoking their
-keys. Further, maintainers may reset the states of the affected refs and sign
-new RSL entries with corrected states.
-
-TODO: evaluate if consecutive state verification fails on clients behind the
-times. Should recovery rewrite non-valid RSL entries? Defeats the purpose?
+gittuf presents mechanisms that can be used in many different scenarios.
```

Comment:
gittuf uses two basic mechanisms for recovery.  We describe these core building blocks for recovery before we discuss the exact scenarios when they are applied and why they provide the desired security properties.